lockc/src/main.rs

```diff
@@ -0,0 +1,179 @@
+                pid,
+                policy_level,
+                responder_tx,
+            } => {
+                let res = add_container(&mut bpf, container_id, pid, policy_level);
+                match responder_tx.send(res) {
+                    Ok(_) => {},
+                    Err(e) => error!("could not send a response for add_container eBPF command to fanotify thread {:?}", e),
+                }
+            }
+            EbpfCommand::DeleteContainer {
+                container_id,
+                responder_tx,
+            } => {
+                let res = delete_container(&mut bpf, container_id);
+                match responder_tx.send(res) {
+                    Ok(_) => {},
+                    Err(e) => error!("could not send a response for delete_container eBPF command to fanotify thread {:?}", e),
+                }
+            }
+            EbpfCommand::AddProcess {
+                container_id,
+                pid,
+                responder_tx,
+            } => {
+                let res = add_process(&mut bpf, container_id, pid);
+                match responder_tx.send(res) {
+                    Ok(_) => {},
+                    Err(e) => error!("could not send a response for add_process eBPF command to fanotify thread {:?}", e),
+                }
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn main() -> Result<()> {
+    let log_level = match env::var("LOCKC_DEBUG") {
+        Ok(_) => LevelFilter::Debug,
+        Err(_) => LevelFilter::Info,
+    };
+    TermLogger::init(
+        LevelFilter::Debug,
+        ConfigBuilder::new()
+            .set_target_level(log_level)
+            .set_location_level(log_level)
+            .build(),
+        TerminalMode::Mixed,
+        ColorChoice::Auto,
+    )?;
+
+    // Step 1: Create a synchronous thread which takes care of fanotify
+    // polling on runc binaries. We monitor all possible runc binaries to get
+    // all runc execution events (and therefore - all operations on
+    // containers).
+    // This thread has to be synchronous and cannot be a part of Tokio runtime,
+    // because it:
+    // * uses the poll() function
+    // * blocks the filesystem operations on monitored files
+    // * in case of monitoring runc, we have to be sure that we register a new
+    //   container exactly before we allow runc to be actually executed;
+    //   otherwise we cannot guarantee that lockc will actually enforce
+    //   anything on that container.
+
+    // Fanotify thread bootstrap channel - used later to start the real bootstrap
+    // of the thread. We want to bootstrap it later, after loading eBPF
+    // programs (which happens in async code in Tokio runtime).
+    let (fanotify_bootstrap_tx, fanotify_bootstrap_rx) = oneshot::channel::<()>();
+
+    // eBPF thread channel - used by fanotify thread to request eBFP operations
+    // from the async eBPF thread.
+    let (ebpf_tx, ebpf_rx) = mpsc::channel::<EbpfCommand>(32);
```

Comment:
Isn't the size of this channel too small (`32`)? I have no idea about the amount of requests that could be generated...

Reply:
Good point, I will increase it to 100